Spigot-Server-Patches/0151-Basic-PlayerProfile-API.patch

```diff
@@ -178,7 +178,7 @@ index 0000000000000000000000000000000000000000..2751ce7f1556da07ef853807a588f096
 +
 +        if ((profile.getName() == null || !hasTextures()) && profile.getId() != null) {
 +            GameProfile profile = userCache.getProfile(this.profile.getId());
-+            if (profile != null) {
++            if (profile != null && (this.profile.getName() == null || profile.getProperties().containsKey("textures"))) {
```

Comment:
doesn't feel like the correct solution.

Feels like we should update this.profile only if name is null, and then copy profile into this.profile instead.

so that if you already had name and uuid on this.profile, we only import properties, but if this.profile is missing name, but user cache profile had name, then we just swap references.

Reply:
So something like:
```java
if (profile != null) {
    if (this.profile.getName == null) {
        // if old has it, assume its newer, so overwrite, else use cached if it was set and ours wasn't
        copyProfileProperties(this.profile, profile);
        this.profile = profile;
    } else {
        copyProfileProperties(profile, this.profile);
    }
}```
?

Reply:
I think so.

Reply:
Addressed.